From 978ce0da35d3447e6c285c78f1f8ad56c4c26400 Mon Sep 17 00:00:00 2001
From: Stefan Eichenberger <stefan.eichenberger@toradex.com>
Date: Wed, 18 Mar 2020 18:59:00 +0100
Subject: [PATCH 2/3] arm64: dts: fsl-imx8qxp-apalis: add configuration for
 torizon

For torizon we disable all interfaces which are not part of the module.

Signed-off-by: Stefan Eichenberger <stefan.eichenberger@toradex.com>
---
 .../dts/freescale/fsl-imx8qxp-apalis-eval.dts |  1 +
 .../freescale/fsl-imx8qxp-apalis-torizon.dtsi | 22 +++++++++++++++++++
 2 files changed, 23 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-torizon.dtsi

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-eval.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-eval.dts
index 79bdbf73ff75f..742bb1c90de39 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-eval.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-eval.dts
@@ -7,6 +7,7 @@
 
 #include "fsl-imx8qxp-apalis.dtsi"
 #include "fsl-imx8qxp-apalis-eval.dtsi"
+#include "fsl-imx8qxp-apalis-torizon.dtsi"
 
 / {
 	model = "Toradex Apalis iMX8QXP/DX on Apalis Evaluation Board";
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-torizon.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-torizon.dtsi
new file mode 100644
index 0000000000000..125f6f718b0b2
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-apalis-torizon.dtsi
@@ -0,0 +1,21 @@
+/ {
+    lvds1_panel {
+        status = "disabled";
+    };
+};
+
+/delete-node/ &atmel_mxt_ts;
+
+&adma_lcdif {
+    status = "disabled";
+};
+
+&i2c0 {
+    /delete-node/ ad7879@2c;
+};
+
+/delete-node/ &pinctrl_ad7879_int;
+
+&ldb1 {
+    status = "disabled";
+};
-- 
2.17.1

